### DISCUSSION TITLE
The signature of `Callback.on_train_epoch_end` has changed i

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Lightning-AI/pytorch-lightning#8669] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am getting this warning. 
```
ightningDeprecationWarning: The signature of `Callback.on_train_epoch_end` has changed in v1.3. `outputs` parameter has been removed. Support for the old signature will be removed in v1.5 "The signature of `Callback.on_train_epoch_end` has changed in v1.3."
```
What does that mean? and how to avoid this

**Comment by ananthsub** [comment]
Have you implemented a callback which implements the `on_train_epoch_end` method? Does it contain an `outputs` parameter in the method signature? 

If so, you should remove `outputs` as the callback method. if you need the per-step outputs to post-process in the callback, see here for some options: https://github.com/PyTorchLightning/pytorch-lightning/blob/e5d9e21dea0caa625f26f5cec169195ecbe5e930/pytorch_lightning/callbacks/base.py#L112-L120

**Reply by talhaanwarch** [reply]
I am not using `on_train_epoch_end ` method
Here is my function
```
def validation_step(self,batch,batch_idx):
   
    self.log('val_dice', dice, on_step=False, on_epoch=True,prog_bar=True)
    self.log('val_loss', loss, on_step=False, on_epoch=True,prog_bar=True)
    return loss
```
